docs/styleguide/PyGuide.md

```diff
@@ -0,0 +1,820 @@
+  from foo.bar.yourclass import YourClass
+  ```
+  If this spelling causes local name clashes, then spell them explicitly:
+  ```python
+  import myclass
+  import foo.bar.yourclass
+  ```
+  and use `myclass.MyClass` and `foo.bar.yourclass.YourClass`.
+
+- Wildcard imports (from <module> import *) should be avoided, as they make it unclear which names are present 
+  in the namespace, confusing both readers and many automated tools.
+
+<a id="s2.3-3rd-party-packages"></a>
+<a id="23-3rd-party-packages"></a>
+<a id="3rd-party-packages"></a>
+### 2.3 3rd party packages 
+
+Do not add new third-party dependencies unless absolutely necessary. All things being equal, give preference to built-in packages.  
+
+<a id="s2.4-global-variables"></a>
+<a id="24-global-variables"></a>
+<a id="global-variables"></a>
+### 2.4 Global variables 
+
+Avoid global variables.
+
+- Module-level constants are permitted and encouraged. For example: `MAX_HOLY_HANDGRENADE_COUNT = 3`. Constants must be 
+  named using all caps with underscores.
+- If needed, globals should be declared at the module level and made internal to the module by prepending an `_` to the 
+  name. External access must be done through public module-level functions.
+
+<a id="s2.5-nested"></a>
+<a id="25-nested"></a>
+<a id="nested-classes-functions"></a>
+### 2.5 Nested/Local/Inner Classes and Functions 
+
+No need to overuse nested local functions or classes and inner classes.
+
+- Nested local functions or classes are fine if it satisfy the following conditions:
+  - The code becomes more readable and simpler.
+  - Closing over a local variables.
+  ```python
+  # Correct:
+  def make_scaling_fn(scale):
+      def mul(x):
+          return scale * x
+      return mul
+  ```
+
+- Inner classes are fine when it creates more readble and simple code.
+
+- Do not nest a function just to hide it from users of a module. Instead, prefix its name with an \_ at the module 
+  level so that it can still be accessed by tests.
+  ```Python
+  # Wrong:
+  def avg(a, b, c):
+
+      def sum(x, y):
+          return x + y
+
+      m = sum(a,b)
+      m = sum(m,c)
+      return m/3
+  ```
+  ```Python
+  # Correct:
+  def _sum(x, y):
+      return x + y
+
+  def avg(a, b, c):
+      m = _sum(a,b)
+      m = _sum(m,c)
+      return m/3
+  ```
+
+<a id="s2.6-default-iterators-and-operators"></a>
+<a id="26-default-iterators-and-operators"></a>
+<a id="default-iterators-operators"></a>
+### 2.6 Default Iterators and Operators 
+
+Use default iterators and operators for types that support them, like lists,
+dictionaries, and files. The built-in types define iterator methods, too. Prefer
+these methods to methods that return lists, except that you should not mutate a
+container while iterating over it.
+
+```python
+# Correct:
+for key in adict: ...
+if key not in adict: ...
+if obj in alist: ...
+for line in afile: ...
+for k, v in adict.items(): ...
+for k, v in six.iteritems(adict): ...
```

Comment:
I think we can remove it. `six` is a Python 2 and 3 compatibility library. We use only Python 3.